pages/documentation/developer/tutorials/balance_computation.md

```diff
@@ -0,0 +1,558 @@
+    validationParameters.getOutputDir().ifPresent(outputDir -> {
+    try {
+    System.setOut(new PrintStream(outputDir + "/output-balances-ajustement.log"));
+    } catch (FileNotFoundException e) {
+    e.printStackTrace();
+    }
+    });
+}
+```
+Then we will create a special method to import the IGMs outside the Main method:
+```java
+private static Map<String, Network> importNetworks(BalancesAdjustmentValidationParameters validationParameters) {
+        Map<String, Network> networks = new HashMap<>();
+        validationParameters.getIgmPaths().forEach((name, path) -> networks.put(name, Importers.loadNetwork(Paths.get(path))));
+        for (Map.Entry<String, Network> entry : networks.entrySet()) {
+            String name = entry.getKey();
+            Network network = entry.getValue();
+        return networks;
+    }
+```
+The networks are stored in a HashMap. We will call this method in Main to import the networks through:
+```java
+Map<String, Network> networks = importNetworks(validationParameters);
+```
+
+### Power flow on the IGMs
+Then we will compute a power flow on the networks and select the valid ones, those for which the loadflow is successful. 
+We will create a new method for this loadflow preprocessing:
+```java
+private static void loadflowPreProcessing(Map<String, Network> networks, Map<String, Network> validNetworks) {
+    // launch loadflow for each IGM
+    networks.forEach((name, network) -> {
+        LoadFlowResult result = LoadFlow.run(network, LOAD_FLOW_PARAMETERS);
+        System.out.println(name + " loadflow: " + result.isOk());
+        if (!result.isOk()) {
+            validNetworks.remove(name);
+        }
+    });
+}
+```
+And we will add the corresponding code in the main method:
+```java
+Map<String, Network> validNetworks = new HashMap<>(networks);
+if (LOAD_FLOW_PREPROCESSING) {
+    loadflowPreProcessing(networks, validNetworks);
+}
+```
+
+### Merge of the IGMs and power flow on the CGM
+Finally, we will merge the IGMs to create the CGM:
+```java
+MergingView mergingView = MergingView.create("merged", "validation"); 
+mergingView.merge(validNetworks.values().toArray(Network[]::new));
+```
+And run a loadflow on the CGM:
+```java
+if (!PREPARE_BALANCE_COMPUTATION) {
+    LOAD_FLOW_PARAMETERS.setReadSlackBus(false);
+    LOAD_FLOW_PARAMETERS.setDistributedSlack(true);
+    LoadFlowResult result = LoadFlow.run(mergingView, LOAD_FLOW_PARAMETERS);
+    for (Generator gen : mergingView.getGenerators()) {
+        gen.setTargetP(-gen.getTerminal().getP()); // optional?
+    }
+    System.out.println(result.isOk());
+    System.out.println(result.getMetrics());
+}
+```
+
+### Balance computation
+Now we will start the balance computation. Indeed, the PEVF input gives the actual net position of each IGM that has been merged. As it it is market data, this net position is the one who should obtain after the lodflow, it is the target net position.
```

Comment:
You have to mention that the power factor remains constant during scaling. You also have to speak about the fictitious area with dangling lines to adapt the boundary lines when the global net position differs from the global target. For internal scaling, only loads are used.